### ISSUE TITLE
Support nested fields

### ISSUE THREAD

**Issue by curquiza** [meilisearch/meilisearch#2211]
Related to this spec https://github.com/meilisearch/specifications/pull/121/files that describes the different data types supported for the fields in a document and how Meilisearch handles them.

🔥 Please refer to the spec for any implementation 🔥 

----

# TLDR;

Meilisearch will support nested fields in the documents.

```json
{
    "id": 0,
    "person": {
         "firstname": "Guillaume"
    }
}
```

Is internally represented

```json
{
    "id": 0,
    "person.firstname": "Guillaume"
}
```

It makes it possible for a user to express a nested field location in any API parameter accepting document attributes using the `.` (dot-notation).
It can be expressed this way in any API parameter accepting document attributes. 

e.g. `"sortableAttributes": ["person.firstname"]`

and will be able to process search and to

**Comment by curquiza** [comment]
@meilisearch/integration-team should check in their test suite they can support nested fields :)

**Comment by jkdoshi** [comment]
Question: Will it also allow indexing/filtering/sorting of values within arrays?

For example:

```json
{
  "prices": [
    {"price":22,"per":"kg"},
    {"price":10,"per":"lb"},
  ]
}
```

And then filter or sort on `prices[0].price`?

**Comment by MarinPostma** [comment]
@jkdoshi afaik this is not supported, but this will `prices.price == 10 AND prices.per == "kg"`. Get me wrong @irevoire ?

**Comment by irevoire** [comment]
Yes, this is not planned. Currently, when you filter on an array, you can’t select one value, and that’ll be the same with an array of objects.
Your solution will be to do as Marin said 👍

**Comment by curquiza** [comment]
Closed by https://github.com/meilisearch/meilisearch/pull/2298
